src/Voron/StorageEnvironment.cs

```diff
@@ -1453,8 +1493,22 @@ internal class TestingStuff
         {
             internal Action ActionToCallDuringFullBackupRighAfterCopyHeaders;
         }
+
+        private class EnvironmentWeakReference
```

Comment:
Can be replaced with `WeakReference<T>`, no?

Reply:
No, because `WeakReference<T>` give you only the option to get the target object with the usage of `TryGetTarget(out target)` which forces the allocation of local variable which is copied to the state machine